quic/s2n-quic-core/src/frame/mod.rs

```diff
@@ -36,7 +36,7 @@ pub trait FrameTrait: AckElicitable + CongestionControlled + path_validation::Pr
 impl<T: AckElicitable + CongestionControlled + path_validation::Probing> FrameTrait for T {}
 
 macro_rules! frames {
-    ($ack:ident, $data:ident | $($tag_macro:ident => $module:ident, $handler:ident, $ty:ident $([$($generics:tt)+])?;)*) => {
+    ($ack:ident, $data:ident | $($([$tag_macro:ident])? $(extension[$extension_tag_macro:ident])? => $module:ident, $handler:ident, $ty:ident $([$($generics:tt)+])?;)*) => {
```

Comment:
I wonder if instead of splitting the parsers we just do a `peek::<VarInt>()?` instead of `peek::<u8>()?` and then you can match on `tag.as_u64()`

Reply:
I didn't go down that route since the RFC calls out the efficiency of using a single byte, would peeking a `VarInt` be any less efficient?

> To ensure simple and efficient implementations of frame parsing, a frame type MUST use the shortest possible encoding. For frame types defined in this document, this means a single-byte encoding

Also, I think this would require changing all the tags on the standard frames to `u64`s?